prediction_market_agent/agents/learnable_agent/functions.py

```diff
@@ -0,0 +1,148 @@
+import typing as t
+from typing import Callable
+
+from autogen import register_function, ConversableAgent
+from pydantic import BaseModel
+from tavily import TavilyClient
+
+from prediction_market_agent.utils import APIKeys
+
+
+class FunctionDefinition(BaseModel):
+    function_name: str
+    function_callable: Callable
+
+
+tavily = TavilyClient(api_key=APIKeys().tavily_api_key.get_secret_value())
```

Comment:
Don't initialise stuff on the import level please, what happen is that it will error out if `tavily_api_key` isn't defined for some reason in the environment (for example, arbitrage agent and invalid agent come to mind)

Reply:
Moved to inside the function